pkg/mesh/krun.go

```diff
@@ -233,6 +236,51 @@ func (kr *KRun) InitForTD() {
 	}
 }
 
+// LoadTDBootstrapConfigurations loads optional bootstrap configs for TD from environment variables
+func (kr *KRun) LoadTDBootstrapConfigurations() error {
+	envoyPortRaw := os.Getenv("ENVOY_PORT")
+	if len(envoyPortRaw) > 0 {
+		envoyPort, err := strconv.Atoi(envoyPortRaw)
+		if err != nil {
+			return err
+		}
+		if envoyPort < 0 || envoyPort > 65535 {
+			return errors.New("Envoy Port provided is not a valid port")
+		}
+		kr.TdSidecarEnv.EnvoyPort = envoyPortRaw
+	}
+
+	adminPortRaw := os.Getenv("ADMIN_PORT")
+	if len(adminPortRaw) > 0 {
+		adminPort, err := strconv.Atoi(adminPortRaw)
+		if err != nil {
+			return err
+		}
+		if adminPort < 0 || adminPort > 65535 {
+			return errors.New("Admin Port provided is not a valid port.")
+		}
+		kr.TdSidecarEnv.EnvoyAdminPort = adminPortRaw
+	}
+
+	serviceCIDRRaw := os.Getenv("SERVICE_CIDR")
```

Comment:
Please check the env variables in pilot_agent - there is an env variable for this ( and other iptables related), we should use the same.

Can you also sync up with Vishal - he is starting envoy using pilot-agent already.

Reply:
I have removed the redundant flag, DNS interception differs since we're using iptables.sh, will keep that for now.